core/include/jiminy/core/robot/Robot.h

```diff
@@ -138,6 +183,10 @@ namespace jiminy
 
         std::vector<std::string> motorTorqueFieldnames_;    ///< Fieldnames of the torques of the motors
 
+        std::vector<std::tuple<std::string, std::shared_ptr<AbstractConstraint>, int>> constraintsHolder_;
```

Comment:
Don't use `int`, but rather `int32_t`. and use a struct instead of a tuple in order to have named attribute.